restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -702,23 +702,32 @@ public JsonNode searchDocumentsV2(
       DocumentDB db,
       String keyspace,
       String collection,
+      List<PathSegment> pathPrefix,
       List<FilterCondition> filters,
       List<String> fields,
       String documentId,
       Paginator paginator,
       ExecutionContext context)
       throws UnauthorizedException {
     db.authorizeSelect(keyspace, collection);
-    FilterCondition first = filters.get(0);
-    List<String> path = first.getPath();
+    List<String> path;
+    if (filters.isEmpty()) {
+      path =
+          pathPrefix.stream()
```

Comment:
should we do a conversion from http stuff to string and then have a `List<String>` as the argument here? Don't think that the `DocumentService` should know about the `javax.ws.rs.core` package, right?

Reply:
Good point. I missed the package... thought it was our class :facepalm:

Reply:
Yet, `PathSegment` is already imported into `DocumentService` and is used in other methods' params... Is it worth refactoring all of that now? I suppose we are going to keep refactoring `DocumentService` gradually as we move towards OR support. For this PR I'd like to minimize changes. WDYT?

Reply:
Yep also fine..